core/src/edu/cornell/gdiac/main/LevelModel.java

```diff
@@ -657,20 +668,31 @@ public void makeTransparentObject(SimpleObstacle obstacle, AssetDirectory direct
 
 
   public void createFrames(SimpleObstacle obstacle, AssetDirectory directory, String key, int rows,
-      int cols, int frame) {
-    Animator.create(obstacle, directory, key, rows, cols, frame);
+      int cols, int frame, AnimationType type) {
+    Animator.create(obstacle, directory, key, rows, cols, frame, type);
   }
 
-  public void nextFrame(SimpleObstacle obstacle) {
-    Animator.getTextureRegion(obstacle);
+  public void nextFrame(SimpleObstacle obstacle, AnimationType type) {
+    Animator.getTextureRegion(obstacle, type);
   }
```

Comment:
same thing with this, this is the point of the `Animator` class, we don't want the `LevelModel` to be responsible for this